src/main/java/se/sundsvall/emailreader/service/EmailScheduler.java

```diff
@@ -1,97 +1,51 @@
 import org.springframework.stereotype.Component;
 
-import microsoft.exchange.webservices.data.property.complex.ItemId;
 import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
-import se.sundsvall.emailreader.integration.db.CredentialsRepository;
-import se.sundsvall.emailreader.integration.db.EmailRepository;
-import se.sundsvall.emailreader.integration.db.entity.EmailEntity;
-import se.sundsvall.emailreader.integration.ews.EWSIntegration;
-import se.sundsvall.emailreader.integration.messaging.MessagingIntegration;
-import se.sundsvall.emailreader.service.mapper.EmailMapper;
-import se.sundsvall.emailreader.utility.EncryptionUtility;
 
 @Component
 public class EmailScheduler {
 
-	private static final String EMAIL_SUBJECT = "[Warning] EmailReader has detected unhandled emails";
-
 	private static final Logger log = LoggerFactory.getLogger(EmailScheduler.class);
 
-	private final EWSIntegration ewsIntegration;
-
-	private final MessagingIntegration messagingIntegration;
-
-	private final EmailRepository emailRepository;
-
-	private final CredentialsRepository credentialsRepository;
+	private final EmailService emailService;
 
-	private final EncryptionUtility encryptionUtility;
-
-
-	public EmailScheduler(final EWSIntegration ewsIntegration, final EmailRepository emailRepository,
-		final CredentialsRepository credentialsRepository, final EncryptionUtility encryptionUtility, final MessagingIntegration messagingIntegration) {
-
-		this.ewsIntegration = ewsIntegration;
-		this.emailRepository = emailRepository;
-		this.credentialsRepository = credentialsRepository;
-		this.encryptionUtility = encryptionUtility;
-		this.messagingIntegration = messagingIntegration;
+	public EmailScheduler(final EmailService emailService) {
+		this.emailService = emailService;
 	}
 
+	/**
+	 * This method is scheduled to run at a fixed rate to check for new emails and save them to the
+	 * database. It is also responsible for moving the emails to the processed folder.
+	 */
 	@Scheduled(initialDelayString = "${scheduled.initial-delay}", fixedRateString = "${scheduled.fixed-rate}")
 	@SchedulerLock(name = "checkForNewEmails", lockAtMostFor = "${scheduled.shedlock-lock-at-most-for}")
 	public void checkForNewEmails() {
 		log.info("Checking for new emails");
-
-		credentialsRepository.findAll().forEach(credential -> credential.getEmailAddress().forEach(emailAddress -> {
-
-			final var result = ewsIntegration
-				.pageThroughEntireInbox(credential.getUsername(), encryptionUtility.decrypt(credential.getPassword()), credential.getDomain(), emailAddress);
-
-
-			result.forEach(email -> {
-
-				try {
-					emailRepository.save(EmailMapper.toEmailEntity(email, credential.getNamespace(), credential.getMunicipalityId(), credential.getMetadata()));
-				} catch (final Exception e) {
-					log.error("Failed to save email", e);
-					return;
+		for (final var credential : emailService.getAllCredentials()) {
```

Comment:
This might just be me, but I preffer to split nestled for-loops into separate methods like
```
	public void checkForNewEmails() {
		log.info("Checking for new emails");

		for (final var credential : emailService.getAllCredentials()) {
			processCredential(credential);
		}
	}

	private void processCredential(final CredentialsEntity credential) {
		for (final var address : credential.getEmailAddress()) {
			processAddress(credential, address);
		}
	}

	private void processAddress(final CredentialsEntity credential, final String address) {
		for (final var email : emailService.getAllEmailsInInbox(credential, address)) {
			processEmail(credential, email, address);
		}
	}

	private void processEmail(CredentialsEntity credential, Email email, String address) {
		try {
			emailService.saveAndMoveEmail(email, address, credential);
		} catch (Exception e) {
			log.error("Failed to handle email with subject: {}", email.subject(), e);
		}
	}

```

Reply:
This is very subjective, in this case it is very visible that each for-loop only calls another for-loop. If there was more steps within each for-loop it would get real messy real fast.

Linus suggested using forEach instead to make it more readable, I might use that approach instead.